api/tests/routes/backoffice/finance_test.py

```diff
@@ -894,3 +896,211 @@ def test_generate_invoices_when_task_already_running(self, authenticated_client)
                 html_parser.extract_alert(authenticated_client.get(response.location).data)
                 == "La tâche de génération des justificatifs est déjà en cours"
             )
+
+
+class ForceDebitNoteTest(PostEndpointHelper):
+    endpoint = "backoffice_web.finance_incidents.force_debit_note"
+    endpoint_kwargs = {"finance_incident_id": 1}
+    needed_permission = perm_models.Permissions.MANAGE_INCIDENTS
+
+    def test_force_debit_note(self, authenticated_client):
+        venue = offerers_factories.VenueFactory(pricing_point="self", reimbursement_point="self")
+        booking = bookings_factories.ReimbursedBookingFactory(venue=venue, stock__offer__venue=venue)
+        original_event = finance_factories.UsedBookingFinanceEventFactory(booking=booking)
+        original_pricing = api.price_event(original_event)
+        original_pricing.status = finance_models.PricingStatus.INVOICED
+
+        now = datetime.datetime.utcnow()
+        finance_incident = finance_factories.IndividualBookingFinanceIncidentFactory(
+            booking=booking, incident__status=finance_models.IncidentStatus.VALIDATED, incident__forceDebitNote=False
+        ).incident
+        api._create_finance_events_from_incident(
+            finance_incident.booking_finance_incidents[0], incident_validation_date=now
+        )
+
+        response = self.post_to_endpoint(
+            authenticated_client,
+            finance_incident_id=finance_incident.id,
+        )
+
+        assert response.status_code == 200
+        assert "Une note de débit sera générée à la prochaine échéance." in html_parser.content_as_text(response.data)
+        updated_finance_incident = finance_models.FinanceIncident.query.get(finance_incident.id)
+        assert updated_finance_incident.forceDebitNote == True
+
+    @pytest.mark.parametrize(
+        "incident_status",
+        [
+            finance_models.IncidentStatus.CANCELLED,
+            finance_models.IncidentStatus.CREATED,
+        ],
+    )
+    def test_not_force_debit_note(self, authenticated_client, incident_status):
+        original_finance_incident = finance_factories.FinanceIncidentFactory(status=incident_status)
+
+        response = self.post_to_endpoint(
+            authenticated_client,
+            finance_incident_id=original_finance_incident.id,
+        )
+        redirected_response = authenticated_client.get(response.location)
+        content_text = html_parser.content_as_text(redirected_response.data)
+
+        assert "L'incident doit être au statut 'validé' pour effectuer cette action." in content_text
+        finance_incident = finance_models.FinanceIncident.query.get(original_finance_incident.id)
+        assert finance_incident.forceDebitNote == original_finance_incident.forceDebitNote
```

Comment:
C'est le même objet que celui re-récupéré, il y a de fortes chances qu'il soit aussi mis à jour en cas de changement de valeur, donc mieux vaut vérifier : 
```suggestion
        assert finance_incident.forceDebitNote is False
```